File: src/tools/web-fetch.ts
import { Tool } from './tool';
import { registerTool } from './registry';

export const webFetch: Tool = {
  name: 'web-fetch',
  description: 'Fetch content from a URL',
  schema: { type: 'string', description: 'URL to fetch' },
  handler: async (url: string) => {
    const res = await fetch(url);
    return await res.text();
  }
};

registerTool(webFetch);

File: src-tauri/src/tools/web_fetch.rs
use super::{Tool, ToolMetadata};

pub struct WebFetch;

impl Tool for WebFetch {
    fn metadata(&self) -> ToolMetadata {
        ToolMetadata {
            name: "web-fetch",
            description: "Fetch content from a URL",
        }
    }

    fn handle(&self, input: String) -> anyhow::Result<String> {
        let body = reqwest::blocking::get(&input)?.text()?;
        Ok(body)
    }
}
